pkg/services/folder/folderimpl/folder.go

```diff
@@ -565,6 +755,58 @@ func (s *Service) deduplicateAvailableFolders(ctx context.Context, folders []*fo
 }
 
 func (s *Service) GetParents(ctx context.Context, q folder.GetParentsQuery) ([]*folder.Folder, error) {
+	if s.features.IsEnabledGlobally(featuremgmt.FlagKubernetesFoldersServiceV2) {
+		return s.getParentsFromApiServer(ctx, q)
+	}
+
+	return s.GetParentsLegacy(ctx, q)
+}
+
+func (s *Service) getParentsFromApiServer(ctx context.Context, q folder.GetParentsQuery) ([]*folder.Folder, error) {
```

Comment:
Note: This whole blob can be replaced by this code after it's merged: https://github.com/grafana/grafana/pull/98392